src/commands.rs

```diff
@@ -10,36 +10,48 @@ pub(crate) async fn create<R: Runtime>(
    key: String,
    url: String,
    path: String,
-) -> Result<DownloadRecord> {
+) -> Result<DownloadItem> {
    app.download().create(app.clone(), key, url, path)
 }
 
 #[command]
-pub(crate) async fn list<R: Runtime>(app: AppHandle<R>) -> Result<Vec<DownloadRecord>> {
+pub(crate) async fn list<R: Runtime>(app: AppHandle<R>) -> Result<Vec<DownloadItem>> {
    app.download().list(app.clone())
 }
 
 #[command]
-pub(crate) async fn get<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadRecord> {
+pub(crate) async fn get<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadItem> {
    app.download().get(app.clone(), key)
 }
 
 #[command]
-pub(crate) async fn start<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadRecord> {
+pub(crate) async fn start<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadItem> {
    app.download().start(app.clone(), key)
 }
 
 #[command]
-pub(crate) async fn cancel<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadRecord> {
+pub(crate) async fn cancel<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadItem> {
    app.download().cancel(app.clone(), key)
 }
 
 #[command]
-pub(crate) async fn pause<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadRecord> {
+pub(crate) async fn pause<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadItem> {
    app.download().pause(app.clone(), key)
 }
 
 #[command]
-pub(crate) async fn resume<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadRecord> {
+pub(crate) async fn resume<R: Runtime>(app: AppHandle<R>, key: String) -> Result<DownloadItem> {
    app.download().resume(app.clone(), key)
 }
+
+#[tauri::command(rename_all = "snake_case")]
+pub(crate) async fn is_native<R: Runtime>(_app: AppHandle<R>) -> Result<bool> {
```

Comment:
This command allows the JS API to determine whether to register a plugin listener for iOS or a global listener for Android/desktop. Adding this command was a low-cost alternative to bringing in an external dependency such as the OS [plugin](https://v2.tauri.app/plugin/os-info/).